src/commonMain/kotlin/exchange.dydx.abacus/protocols/ParserProtocol.kt

```diff
@@ -71,3 +76,45 @@ inline fun <reified T> ParserProtocol.asTypedList(list: Any?): List<T>? {
         }
     }
 }
+
+inline fun <reified T> ParserProtocol.asTypedObject(item: Any?): T? {
+    if (item is T) {
+        return item
+    }
+    val itemString = if (item is Map<*, *>) {
+        item.toJson()
+    } else {
+        asString(item)
+    }
+    return if (itemString != null) {
+        try {
+            val json = Json {
```

Comment:
I think we can just cache a singleton `Json` instance.

Reply:
Will change in the follow-ups.